src/ru/geekbrains/ntr_ads07/Graph.java

```diff
@@ -98,22 +99,80 @@ public void bfs(String startLabel) {
         Queue<Vertex> queue = new LinkedList<>();
         Vertex vertex = vertexList.get(startIndex);
 
+        visitVertexAndDisplay(queue, vertex);
+        while (!queue.isEmpty()) {
+            vertex = getNearUnvisitedVertex(queue.peek());
+            if (vertex != null) {
+                visitVertexAndDisplay(queue, vertex);
+            } else {
+                queue.remove();
+            }
+        }
+
+        resetVertexState();
+    }
+
+    @Override
+    public void shortestRoute(String startLabel, String endLabel) {
+        int startIndex = indexOf(startLabel);
+        if (startIndex == -1) {
+            throw new IllegalArgumentException("Invalid start label.");
+        }
+
+        int endIndex = indexOf(endLabel);
+        if (endIndex == -1) {
+            throw new IllegalArgumentException("Invalid end label.");
+        }
+
+        if (startIndex == endIndex) {
+            System.out.println("Start label is equals to end label.");
+            return;
+        }
+
+        Queue<Vertex> queue = new LinkedList<>();
+        Vertex vertex = vertexList.get(startIndex);
+
         visitVertex(queue, vertex);
         while (!queue.isEmpty()) {
             vertex = getNearUnvisitedVertex(queue.peek());
             if (vertex != null) {
                 visitVertex(queue, vertex);
+                if (vertex.getLabel().equals(endLabel)) {
+                    break;
+                }
             } else {
                 queue.remove();
             }
         }
-
+        if (queue.isEmpty()) {
+            System.out.printf("The path between %s and %s was not found.\n",
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.